packages/client-search/src/types/SearchResponse.ts

```diff
@@ -1,42 +1,175 @@
 import { ObjectWithObjectID } from '.';
 
-// @todo add docs here.
 export type SearchResponse<TObject = {}> = {
+  /**
+   * The hits returned by the search.
+   *
+   * Hits are ordered according to the ranking or sorting of the index being queried.
+   */
   readonly hits: ReadonlyArray<TObject & ObjectWithObjectID>;
+
+  /**
+   * Index of the current page (zero-based).
+   */
   readonly page: number;
+
+  /**
+   * Number of hits returned (used only with offset)
+   */
   readonly length?: number;
```

Comment:
we could have a discriminated union here: `baseParams & (hitsPerPageParams | offsetParams)`

Reply:
I don't understand your comment.

Reply:
` { hits, ...} & ({ page, hitsPerPage} | { offset, length })`

something like that

Reply:
I won't have any type like that yet. Let's leave it for maybe next time.